storage/queue.go

```diff
@@ -111,6 +111,44 @@ type PeekMessageResponse struct {
 	MessageText    string `xml:"MessageText"`
 }
 
+// QueueMetadata represents queue properties on
+// a specific queue.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dd179384.aspx
+type QueueMetadata struct {
+	ApproximateMessageCount int
+}
+
+// GetMetadata operation retrieves queue
+// properties on the specified queue.
+//
+// See https://msdn.microsoft.com/en-us/library/azure/dd179384.aspx
+func (c QueueServiceClient) GetMetadata(name string) (QueueMetadata, error) {
+	qm := QueueMetadata{}
+	uri := c.client.getEndpoint(queueServiceName, pathForQueue(name), url.Values{"comp": []string{"metadata"}})
+	headers := c.client.getStandardHeaders()
+	resp, err := c.client.exec("GET", uri, headers, nil)
+	if err != nil {
+		return qm, err
+	}
+	defer resp.body.Close()
+
+	for k, v := range resp.headers {
+		if len(v) != 1 {
+			return qm, fmt.Errorf("Unexpected number of valuus (%d) in response header '%s'", len(v), k)
+		}
+
+		if k == approximateMessagesCountHeader {
```

Comment:
how do you know the uppercase/lowercase will match? Server may send lowercase headers or `X-ms-approximate-messages-count` which is ≠ `X-Ms-Approximate-Messages-Count`.